lib/syscall_shim/uk_syscall_binary.c

```diff
@@ -147,6 +153,12 @@ void ukplat_syscall_handler(struct uk_syscall_ctx *usc)
 #endif /* !CONFIG_LIBUKCONSOLE */
 #endif /* CONFIG_LIBSYSCALL_SHIM_STRACE */
 
+	/* Mark system call exit */
+	uk_syscall_exittab_run(&(struct uk_syscall_exit_ctx){
+				.execenv = execenv,
+			  });
+	uk_syscall_nested_depth--;
```

Comment:
Since the counter is in the TLS, any syscalls that clear the TLS of the current thread, (right now only `execve()` comes to mind) will end up with a negative value here. You need to register a handler for `POSIX_PROCESS_EXECVE_EVENT` to set the counter to the right value.

Other syscalls that may wipe the TLS as a result of `clone()` would do that on the child, so I think they're out of context.

Reply:
The wrappers decrement before running the table, so this misplacement here disappears.